packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts

```diff
@@ -598,35 +606,26 @@ export async function checkBurn (
   }
 
   let nearBurnReceipt
-  const nep141Factory = options.nep141Factory ?? getBridgeParams().nep141Factory
-  try {
-    if (transfer.symbol !== 'ETH') {
-      nearBurnReceipt = await parseNep141BurnReceipt(nearBurnTx, nep141Factory, nearProvider)
-    } else {
-      // Withdraw ERC-20 ETH from a silo which doesn't use ETH as native currency.
-      nearBurnReceipt = await parseETHBurnReceipt(nearBurnTx, transfer.auroraEvmAccount ?? 'aurora', nearProvider)
-    }
-  } catch (e) {
-    if (e instanceof TransferError) {
-      return {
-        ...transfer,
-        errors: [...transfer.errors, e.message],
-        status: status.FAILED
-      }
-    }
-    // Any other error like provider connection error should throw
-    // so that the transfer stays in progress and checkWithdraw will be called again.
-    throw e
+  if (transfer.symbol !== 'ETH') {
+    const nep141Factory = options.nep141Factory ?? bridgeParams.nep141Factory
+    nearBurnReceipt = await parseNep141BurnReceipt(nearBurnTx, nep141Factory, nearProvider)
+  } else {
+    // Withdraw ERC-20 ETH from a silo which doesn't use ETH as native currency.
+    // @ts-expect-error
+    const txBlock = await nearProvider.block({ blockId: nearBurnTx.transaction_outcome.block_hash })
+    const blockHeight = Number(txBlock.header.height)
+    const etherNep141FactoryMigrationHeight = options.etherNep141FactoryMigrationHeight ?? bridgeParams.etherNep141FactoryMigrationHeight
+    const etherNep141Factory = blockHeight >= etherNep141FactoryMigrationHeight
+      ? (options.etherNep141Factory ?? bridgeParams.etherNep141Factory)
+      : bridgeParams.auroraEvmAccount
+    nearBurnReceipt = await parseETHBurnReceipt(nearBurnTx, etherNep141Factory, nearProvider)
```

Comment:
The same code is used in the recover function. Can we extract it into a separate function?